### PR TITLE
SCB-1478 support API gateway

### PR DESCRIPTION
**Pull request by tianxiaoliang**

### CODE REVIEW

File: proxy/config/struct_ingress.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package config
```
Comment (WillemJiang) [inline proxy/config/struct_ingress.go]: License header

File: proxy/config/struct_ingress_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package config_test
```
Comment (WillemJiang) [inline proxy/config/struct_ingress_test.go]: License header

File: proxy/pkg/egress/egress_test.go [hunk 3]
```diff
@@ -36,9 +40,21 @@ import (
 func BenchmarkMatch(b *testing.B) {
 	lager.Initialize("", "DEBUG", "",
 		"size", true, 1, 10, 7)
-
-	gopath := os.Getenv("GOPATH")
-	os.Setenv("CHASSIS_HOME", gopath+"/src/github.com/apache/servicecomb-mesher")
+	chassis := []byte(`
+cse:
```
Comment (WillemJiang) [inline proxy/pkg/egress/egress_test.go]: Do we still need to use cse setting here?

File: proxy/protocol/http/gateway.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
+package http
```
Comment (WillemJiang) [inline proxy/protocol/http/gateway.go]: License header.

File: proxy/pkg/egress/egress_test.go (hunk 3 above)
Comment (tianxiaoliang) [reply proxy/pkg/egress/egress_test.go]: yes, it is not changed yet, will support servicecomb prefix in future release

**Comment by WillemJiang** [comment]
@tianxiaoliang The code coverage is dropped , can you take a look at it?

**Comment by tianxiaoliang** [comment]
> @tianxiaoliang The code coverage is dropped , can you take a look at it?

I will add UT in next commit

**Pull request merged**